analysis.py

```diff
@@ -0,0 +1,207 @@
+    #ac.add_account_plugin(deltachat.events.FFIEventLogger(ac))
+    ac.add_account_plugin(plugin)
+    ac.set_config("addr", entry["addr"])
+    ac.set_config("mail_pw", entry["app_pw"])
+
+    for name in ("send_server", "mail_server"):
+        val = entry.get(name)
+        if val is not None:
+            ac.set_config(name, val)
+
+    ac.set_config("mvbox_move", "0")
+    ac.set_config("mvbox_watch", "0")
+    ac.set_config("sentbox_watch", "0")
+    ac.set_config("bot", "1")
+    configtracker = ac.configure()
+    configtracker.wait_finish()
+    ac.start_io()
+    duration = time.time() - begin
+    print("%s: successful login as %s in %.1f seconds." % (entry["addr"], plugin.name, duration))
+    if plugin is not EchoPlugin:
+        output.submit_login_result(entry.get("addr"), duration)
+    ac.output = output
+    return ac
+
+
+def check_account_with_spider(spac: deltachat.Account, account: deltachat.Account):
+    """Send a message to spider and check if a reply arrives.
+
+    :param spac: the deltachat.Account object of the spider.
+    :param account: a deltachat.Account object to check.
+    """
+    chat = account.create_chat(spac)
+    chat.send_text("Begin: %f" % time.time())
+
+
+def parse_config_line(line: str):
+    """Parse config file line and create an entry out of it.
+
+    :param line: a string with several config options separated by spaces
+    :return: a dictionary with account credentials, or None
+    """
+    if line[0] == "#" or line[0] == "\n":
+        return
+    entry = { "line" : line }
+    parameters = line.rstrip().split(" ")
+    for p in parameters:
+        argument = p.partition("=")
+        if argument[1] == "":
+            # no = in parameter
+            continue
+        entry[argument[0]] = argument[2]
+    return entry
+
+
+def parse_accounts_file(accounts_file: str) -> Tuple[list, dict]:
+    """import and parse accounts-file
+
+    :param accounts_file: (str) path to accounts file
+    :return: a list with test account entry dicts, a dict with the spider entry dict
+    """
+    with open(accounts_file, "r", encoding="UTF-8") as f:
+        lines = f.readlines()
+    credentials = []
+    spider = None
+    for line in lines:
+        entry = parse_config_line(line)
+        if entry is not None:
+            if entry.get("spider") == "true":
+                spider = entry
+            else:
+                credentials.append(entry)
+    return credentials, spider
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-y", "--yes", action="store_true", default=False,
+                        help="always answer yes if prompted")
+    parser.add_argument("-a", "--accounts_file", help="a file containing mail accounts",
+                        default="testaccounts.txt")
+    parser.add_argument("-d", "--data_dir", help="directory for the account data",
+                        default="/tmp/" + "".join(random.choices("abcdef",k=5)))
```

Comment:
Right now temporary data is stored in /tmp - I think we discussed once that this is not optimal, see #4. What are possible alternatives?

Reply:
you could use https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory and let the default here be "None" so you only create the dir when you actually run anything and acccess the "data_dir" variable and then call TemporaryDirectory.  you can specify a prefix so you can better recognize/remove the dirs later.

Reply:
closed in #4 :)